src/ui/wxWidgets/DragBarCtrl.cpp

```diff
@@ -96,78 +95,211 @@ struct _DragbarElementInfo {
+      mainFrame->m_tree->OnDrag(event);
+    }
 #endif
+    return;
   }
-  
-  const CItemData *pci(nullptr), *pbci(nullptr);
-  pci = m_frame->GetSelectedEntry();
-  pbci = m_frame->GetBaseEntry(pci);
 
-  return pci && !pci->IsFieldValueEmpty(DragbarElements[idx].ft, pbci);
+  wxString text = GetText(toolId);
+  if (!text.IsEmpty())
+  {
+    wxTextDataObjectEx dataObj(text);
+    wxDropSource source(dataObj, this);
+    switch (source.DoDragDrop())
+    {
+    case wxDragError:
+      wxLogDebug(_("Error dragging"));
+      break;
+    case wxDragNone:
+      wxLogDebug(_("Nothing happened dragging"));
+      break;
+    case wxDragCopy:
+      wxLogDebug(_("Copied successfully"));
+      break;
+    case wxDragMove:
+      wxLogDebug(_("Moved successfully"));
+      break;
+    case wxDragCancel:
+      wxLogDebug(_("Dragging cancelled"));
+      break;
+    default:
+      wxLogDebug(_("Unexpected result dragging"));
+      break;
+    }
+  }
 }
 
-PasswordSafeFrame *DragBarCtrl::GetBaseFrame() const
+void DragBarCtrl::OnUpdateUI(wxUpdateUIEvent& event)
 {
-  return m_frame;
+  const auto mainFrame = wxGetApp().GetPasswordSafeFrame();
+  wxASSERT(mainFrame && mainFrame->m_tree);
+
+  const auto selection         = mainFrame->GetSelectedEntry();
+  const auto isTreeView        = mainFrame->IsTreeView();
+  const auto hasGroupSelection = mainFrame->m_tree->IsGroupSelected();
+  const auto hasItemSelection  = selection != nullptr;
+  const auto hasAnySelection   = hasItemSelection || hasGroupSelection;
+  const auto hasGroup          = hasItemSelection && (selection->IsGroupSet());
+  const auto hasTitle          = hasItemSelection && (selection->IsTitleSet());
+  const auto hasUser           = hasItemSelection && (selection->IsUserSet());
+  const auto hasPassword       = hasItemSelection && (selection->IsPasswordSet());
+  const auto hasNotes          = hasItemSelection && (selection->IsNotesSet());
+  const auto hasURL            = hasItemSelection && (selection->IsURLSet());
+  const auto hasEmail          = hasItemSelection && (selection->IsEmailSet());
+
+  switch (event.GetId()) {
+    case DRAGBAR_TOOLID_BASE:     // the 'Group' tool item
+      event.Enable(hasGroup);
+      break;
+    case DRAGBAR_TOOLID_BASE + 1: // the 'Title' tool item
+      event.Enable(hasTitle);
+      break;
+    case DRAGBAR_TOOLID_BASE + 2: // the 'User' tool item
+      event.Enable(hasUser);
+      break;
+    case DRAGBAR_TOOLID_BASE + 3: // the 'Password' tool item
+      event.Enable(hasPassword);
+      break;
+    case DRAGBAR_TOOLID_BASE + 4: // the 'Notes' tool item
+      event.Enable(hasNotes);
+      break;
+    case DRAGBAR_TOOLID_BASE + 5: // the 'URL' tool item
+      event.Enable(hasURL);
+      break;
+    case DRAGBAR_TOOLID_BASE + 6: // the 'Email' tool item
+      event.Enable(hasEmail);
+      break;
+    case DRAGBAR_TOOLID_BASE + 7: // the 'DnD' tool item
```

Comment:
How are these offsets (0-7) defined? Could they ever change?

Reply:
Resolved in fc4ed9d